Docs/agent documentation (#57)
eProsima/Micro-XRCE-DDS-docs: docs/transport.rst

    custom_agent.start();

*CustomEndPoint*
****************

> The :code:`custom_endpoint` is an object of type `eprosima::uxr::CustomEndPoint` and it us in charge of handling the endpoint parameters. The *Agent*, unlike the *Client*, can receive

Reviewer: ```suggestion
The :code:`custom_endpoint` is an object of type ``eprosima::uxr::CustomEndPoint`` and it is in charge of handling the endpoint parameters. The *Agent*, unlike the *Client*, can receive
```

> messages from multiple *Clients* so it must be able to differentiate between different *Clients*.

Reviewer: ```suggestion
messages from multiple *Clients* so it must be able to differentiate between them.
```

Therefore, the :code:`eprosima::uxr::CustomEndPoint` should be provided with information about the origin of the message
in the read callback, and with information about the destination of the message in the write callback.

> In general, members of a :code:`eprosima::uxr::CustomEndPoint` object can be unsigned integers and strings.

Reviewer: ```suggestion
In general, the members of a :code:`eprosima::uxr::CustomEndPoint` object can be unsigned integers and strings.
```

> `CustomEndPoint` defines three methods:

Reviewer 2: +100

> `CustomEndPoint` defines three methods:

Reviewer: ```suggestion
``CustomEndPoint`` defines three methods:
```

Add member
    .. code-block:: cpp

        bool eprosima::uxr::CustomEndPoint::add_member<*KIND*>(const std::string& member_name);

>     Allows to dinamically add a new member to the endpoint definition.

Reviewer 2: ```suggestion
    Allows to dynamically add a new member to the endpoint definition.
```

>     Returns `true` if member was correctly added, or `false` if something went wrong (for example, the member already existed).

Reviewer: ```suggestion
    Returns ``true`` if the member was correctly added, ``false`` if something went wrong (for example, if the member already exists).
```

>     :KIND: To be chosen from: `uint8_t`, `uint16_t`, `uint32_t`, `uint64_t`, `uint128_t` or `std::string`.

Reviewer: ```suggestion
    :KIND: To be chosen from: ``uint8_t``, ``uint16_t``, ``uint32_t``, ``uint64_t``, ``uint128_t`` or ``std::string``.
```

    :member_name: The tag used to identify the endpoint member.

Set member value
    .. code-block:: cpp

        void eprosima::uxr::CustomEndPoint::set_member_value(const std::string& member_name, const *KIND* & value);

>     Sets the specific value (numeric or string) for a certain member, which must previously exist in the CustomEndPoint.

Reviewer: ```suggestion
    Sets the specific value (numeric or string) for a certain member, which must previously exist in the ``CustomEndPoint``.
```

    :member_name: The member whose value is going to be modified.

>     :value: The value to be set, of `KIND`: `uint8_t`, `uint16_t`, `uint32_t`, `uint64_t`, `uint128_t` or `std::string`.

Reviewer: ```suggestion
    :value: The value to be set, of `KIND`: ``uint8_t``, ``uint16_t``, ``uint32_t``, ``uint64_t``, ``uint128_t`` or ``std::string``.
```

Get member
    .. code-block:: cpp

        const *KIND* & eprosima::uxr::CustomEndPoint::get_member(const std::string& member_name);

    Gets the current value of the member registered with the given parameter.

>     The retrieved value might be an `uint8_t`, `uint16_t`, `uint32_t`, `uint64_t`, `uint128_t` or `std::string`.

Reviewer: ```suggestion
    The retrieved value might be an ``uint8_t``, ``uint16_t``, ``uint32_t``, ``uint64_t``, ``uint128_t`` or ``std::string``.
```
